app/views/proposals/details/_top_notification.html.haml

```diff
@@ -0,0 +1,6 @@
+.medium-12.top-notification-wrapper
+  .row
+    .medium-12.column
+      = image_tag 'icon-configuration.png'
+      %p
+        This is a beta version of the Request Details page. #{link_to 'Switch to previous design', proposals_revert_design_path} if you prefer.
```

Comment:
I suggested the client-side JS approach only because I thought it would require less code and would not require the one-off route.

The approach you have here is reasonable. My assumption is that this feature is to help in the UI transition and might eventually be excised from the code.